### ISSUE TITLE
panic: Tendermint state.AppHash does not match AppHash after replay. Got BF24CF2F65C0DBDBAF7E971C5060CFE8BF6C7593, expected 86A391168D13AD0D92FDBE459D43399EBF5ADA28

### ISSUE THREAD

**Issue by sven20181031** [CyberMiles/travis#23]
I killed the travis by 'kill -9 pid' and restart it use './travis node start --home $HOME/.travis' .
i got this error: 
[root@wallet release]# ./travis node start --home $HOME/.travis
INFO [10-31|08:56:29.666] Maximum peer count                       ETH=25 LES=0 total=25
INFO [10-31|08:56:29.667] Starting peer-to-peer node               instance=vm/v1.8.12-stable/linux-amd64/go1.10.3
INFO [10-31|08:56:29.667] Allocated cache and file handles         database=/root/.travis/vm/chaindata cache=768 handles=512
INFO [10-31|08:56:29.957] Initialised chain configuration          config="{ChainID: 18 Homestead: 0 DAO: 0 DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Engine: ethash}"
WARN [10-31|08:56:29.957] Ethash used in fake mode 
INFO [10-31|08:56:29.957] Initialising Ethereum protocol           versions="[63 62]" network=18
WARN [10-31|08:56:29.959] Head state missing, repairing chain      number=125008 hash=5edf92…5726e0
INFO [10-31|08:56:37.359] Rewound blockchain to past state         number=6584   hash=56801b…27f50c
INFO [10-31|08:56:37.360] Loaded most recent local header          number=125008 hash=5edf92…5726e0 td=10648535633627101836397465359930935556284
INFO [10-31|08:56:37.360] Loaded most recent local full block      number=6584   hash=56801b…27f50c td=799600390751288
INFO [10-31|08:56:37.360] Loaded most recent local fast block      number=125008 hash=5edf92…5726e0 td=10648535633627101836397465359930935556284
INFO [10-31|08:56:37.360] Loaded local transaction journal         transactions=0 dropped=0
INFO [10-31|08:56:37.360] Regenerated local transaction journal    transactions=0 accounts=0
WARN [10-31|08:56:37.360] Blockchain not empty, fast sync disabled 
INFO [10-31|08:56:37.360] Starting P2P networking 
INFO [10-31|08:56:37.361] RLPx listener up                         self="enode://46714d95b699a67f5f6c678ead70f5fadd05e6dd7f99198f2e13d852df1ddfeac47259c22a187a47a4608286bb706fded769203c3da8d71ba3596abd4f03b29d@[::]:30303?discport=0"
INFO [10-31|08:56:37.364] IPC endpoint opened                      url=/root/.travis/geth.ipc
INFO [10-31|08:56:37.365] HTTP endpoint opened                     url=http://localhost:8545  cors= vhosts=localhost
panic: Tendermint state.AppHash does not match AppHash after replay. Got BF24CF2F65C0DBDBAF7E971C5060CFE8BF6C7593, expected 86A391168D13AD0D92FDBE459D43399EBF5ADA28

goroutine 1 [running]:
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus.checkAppHash(0xc42b9fb720, 0xa, 0x1e850, 0x2b70, 0xc424a73da0, 0x14, 0x14, 0x1, 0xc424a73de0, 0x14, ...)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus/replay.go:426 +0x1c7
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(0xc4200ed7c0, 0xc42b9fb720, 0xa, 0x1e850, 0x2b70, 0xc424a73da0, 0x14, 0x14, 0x1, 0xc424a73de0, ...)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus/replay.go:328 +0x3ef
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus.(*Handshaker).Handshake(0xc4200ed7c0, 0x19ba5c0, 0xc42c8aec80, 0x4, 0x19ad580)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/consensus/replay.go:246 +0x3e5
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/proxy.(*multiAppConn).OnStart(0xc42c8aec80, 0xc424a99780, 0x15)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/proxy/multi_app_conn.go:108 +0x527
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/libs/common.(*BaseService).Start(0xc42c8aec80, 0x0, 0x0)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/libs/common/service.go:130 +0x3bd
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/node.NewNode(0xc420cbc480, 0x19af480, 0xc42d14f4d0, 0x1998b20, 0xc42e179620, 0xc42f7c2b30, 0x18636d0, 0x18636d8, 0x19ad580, 0xc42e1799a0, ...)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/node/node.go:187 +0x693
github.com/CyberMiles/travis/server/commands.startTendermint(0x19b5a80, 0xc422f8e580, 0xc42093a360, 0xc4469f08c0, 0xc42016ee00)
	/go/src/github.com/CyberMiles/travis/server/commands/services.go:237 +0x278
github.com/CyberMiles/travis/server/commands.startServices(0x7ffed79fc678, 0xd, 0xc42093a360, 0x1e850, 0x1e850, 0xc428c77cc0)
	/go/src/github.com/CyberMiles/travis/server/commands/services.go:67 +0x34b
github.com/CyberMiles/travis/server/commands.start(0x7ffed79fc678, 0xd, 0xc42093a360, 0x20, 0x2710)
	/go/src/github.com/CyberMiles/travis/server/commands/start.go:100 +0x7a
github.com/CyberMiles/travis/server/commands.startCmd.func1(0xc420176a00, 0xc4201b1b40, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/CyberMiles/travis/server/commands/start.go:75 +0x37d
github.com/CyberMiles/travis/vendor/github.com/spf13/cobra.(*Command).execute(0xc420176a00, 0xc4201b1ac0, 0x2, 0x2, 0xc420176a00, 0xc4201b1ac0)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/CyberMiles/travis/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x24748a0, 0x16104d8, 0x24, 0xc4207bff07)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/CyberMiles/travis/vendor/github.com/spf13/cobra.(*Command).Execute(0x24748a0, 0xc4201b1980, 0x2474990)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/libs/cli.Executor.Execute(0x24748a0, 0x1864378, 0x2, 0xc4207bff20)
	/go/src/github.com/CyberMiles/travis/vendor/github.com/tendermint/tendermint/libs/cli/setup.go:89 +0x4e
main.main()
	/go/src/github.com/CyberMiles/travis/cmd/travis/main.go:48 +0x132

It takes me a lot of time to synchronize nodes (more than 2 days), I dont  know how to fix it, pls help, thanks a lot.

**Comment by longzhi** [comment]
Hi sven20181031, have you solved this problem?

**Comment by maithilikoinex** [comment]
Is this solved, if yes how?

**Comment by maithilikoinex** [comment]
Sync from the most recent snapshot instead

**Comment by tpmccallum** [comment]
I have also received this error today.
I am going to perform a fresh installation of the testnet using the shell script (which is really easy and fast).
https://github.com/CyberMiles/travis#automated-script---cybermiles-testnet
I will let you know how it goes.

**Comment by tpmccallum** [comment]
Ok, so the fresh installation produces the following result
```
Checking connectivity... done.
INFO [11-20|00:35:59.110] Successfully init travis database and create tables! 
INFO [11-20|00:35:59.110] Allocated cache and file handles         database=/home/ubuntu/.travis/vm/chaindata cache=16 handles=16
INFO [11-20|00:35:59.116] Writing custom genesis block 
INFO [11-20|00:35:59.116] Persisted trie from memory database      nodes=3 size=423.00B time=40.193µs gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
INFO [11-20|00:35:59.117] successfully wrote genesis block and/or chain rule set hash=131599…ff6994
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6575  100  6575    0     0  45005      0 --:--:-- --:--:-- --:--:-- 45034
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4185  100  4185    0     0  27519      0 --:--:-- --:--:-- --:--:-- 27532
INFO [11-20|00:35:59.496] Maximum peer count                       ETH=25 LES=0 total=25
INFO [11-20|00:35:59.496] Starting peer-to-peer node               instance=vm/v1.8.12-stable/linux-amd64/go1.10.3
INFO [11-20|00:35:59.496] Allocated cache and file handles         database=/home/ubuntu/.travis/vm/chaindata cache=768 handles=512
INFO [11-20|00:35:59.569] Initialised chain configuration          config="{ChainID: 19 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: <nil> EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: <nil> Engine: unknown}"
WARN [11-20|00:35:59.569] Ethash used in fake mode 
INFO [11-20|00:35:59.569] Initialising Ethereum protocol           versions="[63 62]" network=19
INFO [11-20|00:35:59.569] Loaded most recent local header          number=0 hash=131599…ff6994 td=64
INFO [11-20|00:35:59.569] Loaded most recent local full block      number=0 hash=131599…ff6994 td=64
INFO [11-20|00:35:59.569] Loaded most recent local fast block      number=0 hash=131599…ff6994 td=64
INFO [11-20|00:35:59.569] Regenerated local transaction journal    transactions=0 accounts=0
INFO [11-20|00:35:59.570] Starting P2P networking 
INFO [11-20|00:35:59.570] RLPx listener up                         self="enode://1d15b1cc7b89c005b046fcad482551631ff8f8be4413f5ccab516dac79305eb466c4be4c40e9575a84c9815222fa3e5d7b517c3bd0381327fbe5951d9e1c8ea0@[::]:30333?discport=0"
INFO [11-20|00:35:59.572] IPC endpoint opened                      url=/home/ubuntu/.travis/cybermiles.ipc
INFO [11-20|00:35:59.573] HTTP endpoint opened                     url=http://0.0.0.0:8545                 cors= vhosts=localhost
E[11-20|00:36:01.877] Dialing failed                               module=p2p addr=a2316e5d291bb3ed73a3fcd51d43b1917bff0f0e@18.220.30.167:26656 err="Error{dial tcp 18.220.30.167:26656: connect: connection refused}" attempts=0
E[11-20|00:36:02.783] Error dialing peer                           module=p2p err="Error{dial tcp 18.220.30.167:26656: connect: connection refused}"
I[11-20|00:36:03.121] Executed block                               module=state height=1 validTxs=1 invalidTxs=0
I[11-20|00:36:03.121] Updates to validators                        module=state updates="[{\"address\":\"\",\"pub_key\":\"ULM6FCazRkdKzVGKj3ByYd/wEjGCGTxiTWPqULMjQ/8=\",\"power\":10},{\"address\":\"\",\"pub_key\":\"3y+Vl6ehf6o9senBjPg+9bvcNr2PEEMKNcL1bCRI2dY=\",\"power\":10},{\"address\":\"\",\"pub_key\":\"GTTYPnycTFo2QCOvs/Lb//PL6f1WOkxiTPTRPLmj6kE=\",\"power\":10},{\"address\":\"\",\"pub_key\":\"u6A3D8i876ImxYDMvxWtG20Ar9PkFSaoRPLNSO9oKsk=\",\"power\":10}]"
INFO [11-20|00:36:03.124] Imported new chain segment               blocks=1 txs=1 mgas=1.483 elapsed=507.793µs mgasps=2920.149 number=1 hash=74485e…7eedf1 cache=7.65kB
I[11-20|00:36:03.130] Committed state                              module=state height=1 txs=1 appHash=3EBADDD128403FAE265340BB96FABAA214D1B1A7
panic: Panicked questionably: Failed to process committed block (2:259A5691CE9643A7F42F2367635417FA8E3A530D): Wrong Block.Header.AppHash.  Expected 3EBADDD128403FAE265340BB96FABAA214D1B1A7, got 53320F5C396D56C226E375B6CDAC5569645DE0EE
```

**Comment by tpmccallum** [comment]
Thank you @longzhi I have both a Docker and non-Docker instances of the testnet working now.

I used option 1
```
https://travis.readthedocs.io/en/latest/connect-testnet.html#snapshot
```
and option 3
```
https://travis.readthedocs.io/en/latest/connect-testnet.html#option-3-the-hard-way-binary-from-genesis
```
from the URL which you provided and all works well.

I did notice that some of the most recent snapshots from the following URL actually reproduce the error.
```
https://s3-us-west-2.amazonaws.com/travis-ss-testnet
```
Therefore, I ended up using the snapshot which is mentioned explicitly in the documentation (just have to wait a few extra minutes for my testnet to sync, but that is fine).

Many thanks
Tim
